lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -613,9 +613,34 @@ void TypeLoweringVisitor::processUsers(Value val, ArrayRef<Value> mapping) {
       sfo.replaceAllUsesWith(repl);
       sfo.erase();
     } else {
-      val.dump();
-      val.getDefiningOp()->getParentOfType<FModuleOp>()->dump();
-      llvm_unreachable("Unknown aggregate user");
+      // This means, we have already processed the user, and it didnot lower its
+      // inputs. This is an opaque user, which will continue to have aggregate
+      // type as input, even after LowerTypes. So, construct the vector/bundle
+      // back from the lowered elements to ensure a valid input into the opaque
+      // op. This only supports Bundle or vector of ground type elements.
+      // Recursive aggregate types are not yet supported.
+
+      // This builder ensures that the aggregate construction happens at the
+      // user location, and the LowerTypes algorithm will not touch them any
+      // more, because LowerTypes was reverse iterating on the block and the
+      // user has already been processed.
+      ImplicitLocOpBuilder b(user->getLoc(), user);
+      // Get the operand number in the op.
+      auto it = llvm::find(user->getOperands(), val);
```

Comment:
Instead of finding use from operands, can we change the loop to iterate `getUses`?

Reply:
I realized its not required, just `replaceUsesofWith` is enough here.